dex-it/src/test/scala/com/wavesplatform/it/sync/orderV3/PercentModeMatcherTestSuite.scala

```diff
@@ -0,0 +1,346 @@
+package com.wavesplatform.it.sync.orderV3
+
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory.parseString
+import com.wavesplatform.account.{KeyPair, PublicKey}
+import com.wavesplatform.common.state.ByteStr
+import com.wavesplatform.it.api.SyncHttpApi._
+import com.wavesplatform.it.api.SyncMatcherHttpApi._
+import com.wavesplatform.it.sync.config.MatcherPriceAssetConfig._
+import com.wavesplatform.it.{MatcherSuiteBase, NTPTime}
+import com.wavesplatform.lang.script.Script
+import com.wavesplatform.lang.script.v1.ExprScript
+import com.wavesplatform.lang.v1.compiler.Terms
+import com.wavesplatform.transaction.Asset.{IssuedAsset, Waves}
+import com.wavesplatform.transaction.assets.exchange.{AssetPair, Order}
+import com.wavesplatform.common.utils.EitherExt2
+import scala.concurrent.duration._
+
+class PercentModeMatcherTestSuite extends MatcherSuiteBase with NTPTime {
+
+  import PercentModeMatcherTestSuite._
+
+  val minMatcherFee = 200000L
+  val price = 100000000L
+  val script: Script = ExprScript(Terms.TRUE).explicitGet()
+  val scriptBase64: String = script.bytes.value.base64
+
+  val aliceAssetBase58: String = node
+    .broadcastIssue(
+      alice,
+      name = "AliceCoin",
+      description = "AliceCoin for matcher's tests",
+      quantity = 9999999999999L,
+      decimals = 8,
+      reissuable = false,
+      fee = smartIssueFee,
+      script = None,
+      waitForTx = true
+    )
+    .id
+  val aliceAsset = IssuedAsset(ByteStr.decodeBase58(aliceAssetBase58).get)
+  val aliceScriptedAssetBase58: String = node
+    .broadcastIssue(
+      alice,
+      name = "AliceSmartAsset",
+      description = "AliceSmartAsset for matcher's tests",
+      quantity = 9999999999999L,
+      decimals = 8,
+      reissuable = false,
+      fee = smartIssueFee,
+      script = Some(scriptBase64),
+      waitForTx = true
+    )
+    .id
+  val aliceScriptedAsset = IssuedAsset(ByteStr.decodeBase58(aliceScriptedAssetBase58).get)
+  val aliceAddress = alice.publicKey.toAddress.stringRepr
+  val bobAddress = bob.publicKey.toAddress.stringRepr
+
+  val matcherPublicKey: PublicKey = matcher.publicKey
+
+  "Matcher" - {
+    "when has percent-mode for fee in config and orders placed" - {
+      "should accept orders with amount/price/spending/receiving assets as matcherFeeAsset" in {
+        val minFeePercent = 10
+        for (percentAssetType <- Seq(
+          "amount",
+          "price",
+          "spending",
+          "receiving"
+        )) {
+          docker.restartNode(node,
+            configWithOrderFeePercent(
+              assetType = percentAssetType,
+              minFeePercent = minFeePercent
+            ))
+          for (amountAsset <- Seq(
+            aliceAsset,
```

Comment:
both assets have 8 decimals, need to check fee measurments in pairs like Waves-Wct with much count of extra zeros in price percent mode